src/cljs/ataru/virkailija/application/view.cljs

```diff
@@ -588,15 +589,18 @@
 (defn application-list-contents [applications]
   (let [selected-key (subscribe [:state-query [:application :selected-key]])
         expanded?    (subscribe [:state-query [:application :application-list-expanded?]])]
-    (fn [applications]
-      (into [:div.application-handling__list
-             {:class (str (when (= true @expanded?) "application-handling__list--expanded")
-                          (when (> (count applications) 0) " animated fadeIn"))}]
-            (for [application applications
-                  :let [selected? (= @selected-key (:key application))]]
-              (if selected?
-                [cljs-util/wrap-scroll-to [application-list-row application selected?]]
-                [application-list-row application selected?]))))))
+    (r/create-class
+      {:component-did-update #(dispatch [:application/scroll-list-to-previously-closed-application])
+       :reagent-render       (fn [applications]
+                               (into [:div.application-handling__list
+                                      {:class (str (when (= true @expanded?) "application-handling__list--expanded")
+                                                   (when (> (count applications) 0) " animated fadeIn"))
+                                       :id    "application-handling-list"}]
+                                     (for [application applications
+                                           :let [selected? (= @selected-key (:key application))]]
+                                       (if selected?
+                                         [cljs-util/wrap-scroll-to [application-list-row application selected?]]
```

Comment:
Olisko tän scrollauksen voinu hoitaa kans tolla `:application/scroll-list-to-previously-closed-application` mekanismilla, jos tarkastelis `:previously-closed-application` tilan lisäks sitä onko joku hakemus valittu?